Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs

```diff
@@ -0,0 +1,338 @@
+        }
+
+        private void OnFailedLobbyCreateOrJoin()
+        {
+            UnblockUIAfterLoadingIsComplete();
+        }
+
+        private void OnCreatedLobby(Lobby r)
+        {
+            m_localLobby.ApplyRemoteData(r);
+            m_localUser.IsHost = true;
+
+            m_LobbyAsyncRequests.BeginTracking(r);
+            m_lobbyContentHeartbeat.BeginTracking();
+
+            switch (m_localLobby.OnlineMode)
+            {
+                case OnlineMode.IpHost:
+                    Debug.Log($"Created lobby with ID: {m_localLobby.LobbyID} and code {m_localLobby.LobbyCode}, at IP:Port {m_localLobby.Data.IP}:{m_localLobby.Data.Port}");
+                    break;
+                case OnlineMode.UnityRelay:
+                    Debug.Log($"Created lobby with ID: {m_localLobby.LobbyID} and code {m_localLobby.LobbyCode}, Internal Relay Join Code{m_localLobby.RelayJoinCode}");
+                    break;
+            }
+
+            m_GameNetPortal.PlayerName = m_localUser.DisplayName;
+
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            switch (m_localLobby.OnlineMode)
+            {
+                case OnlineMode.IpHost:
+                    m_GameNetPortal.StartHost(m_localLobby.Data.IP, m_localLobby.Data.Port);
+                    break;
+
+                case OnlineMode.UnityRelay:
+                    m_GameNetPortal.StartUnityRelayHost(cancellationTokenSource.Token);
+                    break;
+            }
+        }
+
+        private void OnJoinedLobby(Lobby remoteLobby)
+        {
+            m_localLobby.ApplyRemoteData(remoteLobby);
+
+             m_LobbyAsyncRequests.BeginTracking(remoteLobby);
+             m_lobbyContentHeartbeat.BeginTracking();
+
+            m_GameNetPortal.PlayerName = m_localUser.DisplayName;
+
+            switch (m_localLobby.OnlineMode)
+            {
+                case OnlineMode.IpHost:
+                    Debug.Log($"Joined lobby with code: {m_localLobby.LobbyCode}, at IP:Port {m_localLobby.Data.IP}:{m_localLobby.Data.Port}");
+                    break;
+                case OnlineMode.UnityRelay:
+                    Debug.Log($"Joined lobby with code: {m_localLobby.LobbyCode}, Internal Relay Join Code{m_localLobby.RelayJoinCode}");
+                    break;
+            }
+
+
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            switch (m_localLobby.OnlineMode)
+            {
+                case OnlineMode.IpHost:
+                    m_ClientNetPortal.StartClient(m_GameNetPortal, m_localLobby.Data.IP, m_localLobby.Data.Port);
+                    break;
+
+                case OnlineMode.UnityRelay:
+                    m_ClientNetPortal.StartClientUnityRelayModeAsync(m_GameNetPortal, m_localLobby.RelayJoinCode, cancellationTokenSource.Token, OnRelayJoinFailed);
+                    break;
+            }
+        }
+
+        private void OnRelayJoinFailed(string message)
```

Comment:
Any reason all of this is in a UI mediator? There's multiple places in this class that just feels like UIMediator is a god class that does everything "lobby".

Reply:
In general, what was your rule of thumb for "what goes where"? How did you choose which logic went in LobbyUIMediator and which logic went in Requests?

Reply:
OnRelayJoinFailed handles UI-related logic that can happen when you do:
 - create a lobby in relay mode
 - but fail to actually bind to relay for whatever reason

LobbyUIMediator contains UI-related logic and basic invocations for the lobby API from the UI side of things. It must go somewhere, so here it goes.